src/Umbraco.Infrastructure/Services/Implement/CacheInstructionService.cs

```diff
@@ -0,0 +1,459 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Umbraco.Cms.Core.Cache;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.Events;
+using Umbraco.Cms.Core.Logging;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Persistence.Repositories;
+using Umbraco.Cms.Core.Scoping;
+using Umbraco.Cms.Core.Sync;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Core.Services.Implement
+{
+    /// <summary>
+    /// Implements <see cref="ICacheInstructionService"/> providing a service for retrieving and saving cache instructions.
+    /// </summary>
+    public class CacheInstructionService : RepositoryService, ICacheInstructionService
+    {
+        private readonly IServerRoleAccessor _serverRoleAccessor;
+        private readonly CacheRefresherCollection _cacheRefreshers;
+        private readonly ICacheInstructionRepository _cacheInstructionRepository;
+        private readonly IProfilingLogger _profilingLogger;
+        private readonly ILogger<CacheInstructionService> _logger;
+        private readonly GlobalSettings _globalSettings;
+
+        private readonly object _locko = new object();
```

Comment:
This lock object is never used as far as I can tell.

Reply:
No, it's not - have removed.